web/html/src/components/table/HierarchicalTable.tsx

```diff
@@ -0,0 +1,245 @@
+  const getVisibleRows = (treeData: Array<HierarchicalRow>) => {
+    // Find root nodes (nodes without parents or with parents that don't exist in our data)
+    const rootNodes = treeData.filter((row) => {
+      return !row.parentId || !treeData.some((parent) => identifier(parent) === row.parentId);
+    });
+
+    const visibleRows: Array<HierarchicalRow> = [];
+
+    const addVisibleDescendants = (node: HierarchicalRow) => {
+      visibleRows.push(node);
+      // Only process children if this node is expanded
+      if (expandedRows[identifier(node)]) {
+        // Find all direct children of this node
+        const nodeId = identifier(node);
+        const childNodes = treeData.filter((row) => row.parentId === nodeId);
+        // Process each child recursively
+        childNodes.forEach((child) => addVisibleDescendants(child));
+      }
+    };
+    // Start with root nodes and build the visible tree
+    rootNodes.forEach((root) => addVisibleDescendants(root));
+    return visibleRows;
+  };
+
+  // Handle row expansion toggling
+  const toggleRowExpanded = (rowId: string | number) => {
+    setExpandedRows((prev) => ({
+      ...prev,
+      [rowId]: !prev[rowId],
+    }));
+  };
+
+  // Process the tree structure
+  const treeData = buildTreeStructure(data);
+  const visibleRows = getVisibleRows(treeData);
+
+  // Helper function to render cell content based on row properties
+  const renderCellContent = (row: HierarchicalRow, child: React.ReactElement) => {
+    // Get the cell content (either from custom renderer or from data property)
+    const cellContent = child.props.cell ? child.props.cell(row) : row[child.props.columnKey || ""];
+    // Apply special styling for non-leaf or indented nodes
+    const level = row.level || 0;
+    if (!row.isLeaf || level > 0) {
+      return <span style={{ marginLeft: row.isLeaf ? `${level * indentSize}px` : "0" }}>{cellContent}</span>;
+    }
+    // Return plain content for root-level leaf nodes
+    return cellContent;
+  };
+
+  // Type guard to check if element is a Column component
+  const isColumnElement = (element: React.ReactNode): element is React.ReactElement => {
+    if (!React.isValidElement(element)) return false;
+    return element.props && ("columnKey" in element.props || "header" in element.props || "cell" in element.props);
+  };
+
+  // Add expand/collapse indicator to the specified column
+  const enhancedChildren = React.Children.map(children, (child) => {
+    if (!isColumnElement(child)) return child;
+    // Now TypeScript knows this is a valid element with the expected props
+    if (child.props.columnKey === expandColumnKey) {
+      const cellRenderer = (row: any) => {
+        const rowId = identifier(row);
+        const isExpanded = expandedRows[rowId];
+        const level = row.level || 0;
+        const indent = level * indentSize;
+
+        return (
+          <div style={{ display: "flex", alignItems: "center" }}>
```

Comment:
In general, please prefer a `.module.scss` file for styles. Here specifically, you could use Bootstrap utility classes.